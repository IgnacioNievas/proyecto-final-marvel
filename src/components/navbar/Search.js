import React, {useState} from 'react'
import logo from '../img/marvel-logo.png'

const Search = ({search}) => {
    const[text,setText] = useState('')

    const onSearch= (e)=>{
        setText(e.target.value);
        search(e.target.value);
    }

    return (
        <section >
            <form>
            <img src= {logo} alt='marvel' width='50px' />
            <i class="fas fa-search"></i><input type="text"
                placeholder="Find a character.."
                onChange={onSearch}
                value={text} />
                <i class="far fa-star"></i>
            </form>
        </section>
    )
}

export default Search