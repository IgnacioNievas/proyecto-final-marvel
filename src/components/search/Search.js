import React, { useState } from 'react';
import logo from '../../assets/images/marvel-logo.jpeg';
import { SearchStyle } from './SearchStyle';

const Search = ({ search }) => {
	const [text, setText] = useState('');

	const onSearch = (e) => {
		setText(e.target.value);
		search(e.target.value);
	};

	return (
		<SearchStyle>
			<div>
				<img src={logo} alt='marvel' />
				<form>
					<i className='fas fa-search'></i>
					<input
						type='text'
						placeholder='Find a character..'
						onChange={onSearch}
						value={text}
					/>
				</form>
				<button>
					<i className='fas fa-star '></i>
				</button>
			</div>
		</SearchStyle>
	);
};

export default Search;
