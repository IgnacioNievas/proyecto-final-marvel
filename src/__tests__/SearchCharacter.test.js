import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { act, create } from 'react-test-renderer';
import Search from '../components/search/SearchCharacter';
import { BrowserRouter } from 'react-router-dom';
// import userEvent from '@testing-library/user-event';

afterEach(cleanup);

test('<Search />, comprabacion de elementos y actualizacion de props', () => {
	const wrapper = render(
		<BrowserRouter>
			<Search />
		</BrowserRouter>
	);
	wrapper.debug();
	let props;
	const input = screen.getByTestId('input');

	expect(input).toBeInTheDocument();
	expect(input.textContent).toBe('');
	expect(input.textContent).not.toBe('characters');
	expect(input.tagName).toBe('INPUT');
	expect(input.tagName).not.toBe('TEXTAREA');

	const button = screen.getByTestId('button');
	expect(button.tagName).toBe('BUTTON');
	expect(button.tagName).not.toBe('INPUT');
	expect(button).toBeInTheDocument();

	act(() => {
		props = create(
			<BrowserRouter>
				<Search value={1} />
			</BrowserRouter>
		);
	});

	// make assertions on props
	expect(props.toJSON()).toMatchSnapshot();

	// update with some different props
	act(() => {
		props.update(
			<BrowserRouter>
				<Search value={2} />
			</BrowserRouter>
		);
	});

	// make assertions on props
	expect(props.toJSON()).toMatchSnapshot();
});
