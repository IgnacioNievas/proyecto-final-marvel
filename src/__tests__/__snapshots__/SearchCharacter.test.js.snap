// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search />, comprabacion de elementos y actualizacion de props 1`] = `
<section
  className="sc-bdnxRM hAbAwW"
>
  <div>
    <button
      onClick={[Function]}
    >
      <img
        alt="marvel"
        src="marvel-logo.jpeg"
      />
    </button>
    <form
      style={
        Object {
          "width": "70%",
        }
      }
    >
      <i
        className="fas fa-search"
      />
      <input
        data-testid="input"
        onChange={[Function]}
        placeholder="Buscar..."
        type="text"
        value=""
      />
    </form>
    <button
      data-testid="button"
      onClick={[Function]}
    >
      <i
        className="far fa-star fa-2x"
        data-testid="i"
      />
    </button>
  </div>
</section>
`;

exports[`<Search />, comprabacion de elementos y actualizacion de props 2`] = `
<section
  className="sc-bdnxRM hAbAwW"
>
  <div>
    <button
      onClick={[Function]}
    >
      <img
        alt="marvel"
        src="marvel-logo.jpeg"
      />
    </button>
    <form
      style={
        Object {
          "width": "70%",
        }
      }
    >
      <i
        className="fas fa-search"
      />
      <input
        data-testid="input"
        onChange={[Function]}
        placeholder="Buscar..."
        type="text"
        value=""
      />
    </form>
    <button
      data-testid="button"
      onClick={[Function]}
    >
      <i
        className="far fa-star fa-2x"
        data-testid="i"
      />
    </button>
  </div>
</section>
`;
